### ISSUE TITLE
[Tabs] Height defined by the tallest tab

### ISSUE THREAD

**Issue by julianobrasil** [angular/components#8445]
#### Bug, feature request, or proposal:

A way to tell **`MatTabGroup`** that its height should be set to the height of the tab with the greatest height value. 

#### What is the expected behavior?

Having a way (maybe an `@Input` property) to tell **`MatTabGroup`** to set its height to be the same as the **`MatTab`** with the greatest height.

#### What is the current behavior?

Without this feature, the container of **`MatTabGroup`** can potencially change size when you switch tabs. In some situations it's not a desirable behavior:

![image](https://media.giphy.com/media/l1Kdcm2t97WMelPsQ/giphy.gif)

#### What is the use-case or motivation for changing an existing behavior?

Better UX

**Comment by jelbourn** [comment]
Is there any benefit to having this as an API vs. setting an explicit height for the tab group?

**Comment by jelbourn** [comment]
Is there any benefit to having this as an API vs. setting an explicit height for the tab group?

**Comment by julianobrasil** [comment]
Well, I'm having some second thoughts... suposing we have, for example, like the gif I pasted above, a table that can grow inside a tab.
1. If you set an explicit height, you won't let the tab grow with its contents and will see the scrolling bars when needed. 
2. If you do not set its height and let it grow with its content, you will push down any content below the MatTabGroup. And this content would move up and down when you switch among different height tabs.
3. If we have the requested feature, we would avoid this "jumping behavior" of the content bellow MatTabGroup. But it should be optional: 
    - The UX would be terrible if the user switched to a shorter tab and didn't see the content below MatTabGroup because a longer content tab would have pushed it far below the screen (but it seems to me that is not a good idea to put things below a MatTabGroup that can grow indefinitelly)
    - But it makes sense with components like a dialog, which have an action row that will always be in the visible area of the screen and you want to allow the dialog's height to grow to the screen size in different display sizes. Maybe I could use a worst-case design and define a height to fit the smallest desktops screen size to avoid problems in this case, not sure if it would be better or worse than what I did.

Ok... It appears this is not broad-use feature. It looks like to be a problem just with small dialogs when is not so difficult to forseen a height for the MatTabGroup. Right now I'm not sure it's worth the effort to modify the API. It's a nice-to-have feature, not something that is killing designs.

**Comment by willshowell** [comment]
@julianobrasil does `dynamicHeight` help here, at least visually? It should add an animation when the height changes.

Also I think you could accomplish this all with css such that the dialog never grows more than 100vh and anything more requires scrolling in the tab body or dialog body.

I do like the idea of defining the body to the size of the largest tab body, but it wouldn't play great with lazy tabs and, as you mentioned, doesn't offer a great solution for tab bodies that have dynamic content, like tables.

**Comment by julianobrasil** [comment]
@willshowell, I had tried `dynamicHeight`, but it didn't make it much better.

As it is a very simple case, I used a fixed height (320px). It satisfactory solved the problem (but I'll also try out a complete layout change in this case - maybe the main problem is not the lack of this feature, but a bad layout).

> it wouldn't play great with lazy 

Yeah, in cases like that the actual height would have to be recalculated every time any change took place in a tab body.
